mne/beamformer/_dics.py

```diff
@@ -1,27 +1,581 @@
 """Dynamic Imaging of Coherent Sources (DICS)."""
 
-# Authors: Roman Goj <roman.goj@gmail.com>
+# Authors: Marijn van Vliet <w.m.vanvliet@gmail.com>
+#          Britta Westner
+#          Susanna Aro <susanna.aro@aalto.fi>
+#          Roman Goj <roman.goj@gmail.com>
 #
 # License: BSD (3-clause)
 
-from copy import deepcopy
+import numbers
 
 import numpy as np
+from scipy import linalg
 
-from ..utils import logger, verbose, warn
+from ..utils import logger, verbose, warn, deprecated
 from ..forward import _subject_from_forward
 from ..minimum_norm.inverse import combine_xyz, _check_reference
 from ..source_estimate import _make_stc
-from ..time_frequency import csd_fourier, csd_multitaper
-from ._lcmv import _prepare_beamformer_input, _setup_picks, _reg_pinv
+from ..time_frequency import csd_fourier, csd_multitaper, csd_morlet
+from ._lcmv import (_prepare_beamformer_input, _setup_picks, _reg_pinv,
+                    _eig_inv, _check_proj_match, _pick_channels_spatial_filter)
 from ..externals import six
 
 
+deprecation_text = ('This function will be removed in 0.17, please use the '
+                    '`make_dics` and `apply_dics_*` functions instead. These '
+                    'functions use a new implementation of DICS that has been '
+                    'more thoroughly tested.')
+
+
 @verbose
-def _apply_dics(data, info, tmin, forward, noise_csd, data_csd, reg,
-                label=None, picks=None, pick_ori=None, real_filter=False,
-                verbose=None):
-    """Dynamic Imaging of Coherent Sources (DICS)."""
+def make_dics(info, forward, csd, reg=0.05, label=None, pick_ori=None,
+              inversion='single', weight_norm=None, normalize_fwd=True,
+              real_filter=False, reduce_rank=False, verbose=None):
+    """Compute a Dynamic Imaging of Coherent Sources (DICS) spatial filter.
+
+    This is a beamformer filter that can be used to estimate the source power
```

Comment:
This is too long for a top-level description. This can all be copy-pasted to `Notes`.